thrust/cub
# CMake update and associated warning fixes.

- Deprecation warnings are disabled due to use of deprecated CUDA APIs
  such as cudaBindTexture. Filed NVlabs/cub#191 to track this.
- Bump CMake minimum version to 3.15.
- Remove workarounds for old cmake/compilers.
- Switch to CUB_ENABLE_DIALECT_CPPXX instead of CMAKE_CXX_STANDARD.
  - Like Thrust multiconfig, all can be enabled at once.
  - Uses Thrust's settings when Thrust multiconfig is enabled.
- Add CUB_ENABLE_* options to disable portions of the build.
  - HEADER_TESTING, TESTING, EXAMPLES
- Add sm_80, remove 30, 32.
  - When building CUB in Thrust, just use thrust's flags.
- Sanitize variable names in AppendOptionIfAvailable.
  - GCC was failing to detect `-Werror`. Long story short, this fixes it.
- Add cub.compiler_interface target to store warning suppressions, etc.
- Find and use Thrust. We should move the Thrust code into Thrust.
- Check for THOROUGH, QUICK and QUICKER variants of tests, create new
  targets for them when found.
- Remove `examples/block/reduce_by_key.cu`, since it doesn't build and is
  incomplete.
- Disable NPP code paths in test_device_histogram as they don't compile.
  Seems like a Dispatch<int, int, NPP> specialization got removed at some
  point.
- Misc warning cleanups.

Fixes thrust/thrust#1192.
